packages/blocks/src/api/parser.js

```diff
@@ -49,6 +61,89 @@ export const toBooleanAttributeMatcher = ( matcher ) => flow( [
 	( value ) => value !== undefined,
 ] );
 
+/**
+ * Returns true if the source name is one which is guaranteed to return
+ * a string value.
+ *
+ * @param {string} source Source name to test.
+ *
+ * @return {boolean} Whether source is one returning a string.
+ */
+export const isStringSource = ( source ) => STRING_SOURCES.has( source );
+
+/**
+ * Returns true if value is of the given JSON schema type, or false otherwise.
+ *
+ * @see http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.25
+ *
+ * @param {*}      value Value to test.
+ * @param {string} type  Type to test.
+ *
+ * @return {boolean} Whether value is of type.
+ */
+export function isOfType( value, type ) {
+	switch ( type ) {
+		case 'string':
+			return String( value ) === value;
+
+		case 'boolean':
+			return Boolean( value ) === value;
+
+		case 'object':
+			return !! value && value.constructor === Object;
+
+		case 'null':
+			return value === null;
+
+		case 'array':
+			return Array.isArray( value );
+
+		case 'integer':
+		case 'number':
+			return Number( value ) === value;
+	}
+
+	return true;
+}
+
+/**
+ * Returns true if value is of an array of given JSON schema types, or false
+ * otherwise.
+ *
+ * @see http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.25
+ *
+ * @param {*}        value Value to test.
+ * @param {string[]} types Types to test.
+ *
+ * @return {boolean} Whether value is of types.
+ */
+export function isOfTypes( value, types ) {
+	return types.some( ( type ) => isOfType( value, type ) );
+}
+
+/**
+ * Returns true if the given attribute schema describes a value which may be
+ * an ambiguous string.
+ *
+ * Some sources are ambiguously serialized as strings, for which value casting
+ * is enabled. This is only possible when a singular type is assigned to the
+ * attribute schema, since the string ambiguity makes it impossible to know the
+ * correct type of multiple to which to cast.
+ *
+ * @param {Object} attributeSchema Attribute's schema.
+ *
+ * @return {boolean} Whether attribute schema defines an ambiguous string
+ *                   source.
+ */
+export function isAmbiguousStringSource( attributeSchema ) {
+	const { source, type } = attributeSchema;
+
+	return (
+		isStringSource( source ) &&
+		typeof type === 'string'
```

Comment:
Should we check if type === 'string' and not typeof type === 'string'?

Reply:
No, because this is how we differentiate between single and multiple types. I can extract a variable to make this clearer.